flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkConfigBuilder.java

```diff
@@ -340,6 +341,11 @@ protected Configuration build() {
         // Set cluster config
         effectiveConfig.setString(KubernetesConfigOptions.NAMESPACE, namespace);
         effectiveConfig.setString(KubernetesConfigOptions.CLUSTER_ID, clusterId);
+
+        if (FlinkUtils.isZookeeperHAActivated(effectiveConfig)) {
+            effectiveConfig.setString(HighAvailabilityOptions.HA_CLUSTER_ID, clusterId);
```

Comment:
By default, `clusterId` is generated [based on the FlinkDeployment resource name](https://github.com/antonipp/flink-kubernetes-operator/blob/6b0d91fe2c1dc41eeba655337c1d99fedabe3ef8/flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkConfigBuilder.java#L96).

The value of `HighAvailabilityOptions.HA_CLUSTER_ID` is later [used](https://github.com/apache/flink/blob/c57b84921447bb0ade5e1ff77a05ebd8bbbe71b7/flink-runtime/src/main/java/org/apache/flink/runtime/util/ZooKeeperUtils.java#L226) as the namespace for storing Zookeeper ZNodes for each Flink application.

This means that if someone creates two `FlinkDeployment`s with the same name in two different Kubernetes namespaces and these two `FlinkDeployment`s use the same ZK cluster for HA metadata storage, there might be a collision. So I think we should append a random UUID to `HighAvailabilityOptions.HA_CLUSTER_ID` :
```java
effectiveConfig.setString(HighAvailabilityOptions.HA_CLUSTER_ID, clusterId + "-RANDOM-UUID-HERE");
```
Not 100% sure of all the implications of doing that but I think it should be safe enough?

Reply:
In that case I would suggest using the namespace + name combination itself to get a unique resource identifier. That is basically the same logic for Kubernetes HA

Reply:
Unfortunately I couldn't implement this logic because the value of `HA_CLUSTER_ID` is overridden way deep in Flink's code 😞 

For Standalone applications it's overridden here in the Operator's code, so it's easy to change:
https://github.com/apache/flink-kubernetes-operator/blob/bf3e1d44506809fda9ec1ef75308496c7202bbff/flink-kubernetes-standalone/src/main/java/org/apache/flink/kubernetes/operator/standalone/KubernetesStandaloneClusterDescriptor.java#L140

However, for Native applications, it's overridden in Flink's code:
https://github.com/apache/flink/blob/ab3eb40d920fa609f49164a0bbb5fcbb3004a808/flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesClusterDescriptor.java#L255

Since this requires changing logic in the main Flink project, I didn't touch the `HA_CLUSTER_ID` value in this Operator PR. It's probably worth doing as a separate ticket later.